WordPress/gutenberg
# wp-env: Add phpMyAdmin support

## What?

Adds phpMyAdmin as an optional service to wp-env.

<img width="1037" alt="wp-env-phpmyadmin" src="https://github.com/user-attachments/assets/260b6b70-b2ae-4389-b837-bc1d7b874c5f">

## Why?

Makes it easier for developers to inspect their wp-env databases, at very little cost.

## How?

- Adds 2 docker-compose services named `phpmyadmin` and `tests-phpmyadmin`.
- ~Adds a switch to `wp-env start` (`--phpmyadmin`) to opt in to the service.~
- Adds extension points in the form of a `.wp-env.json` environment config key (`phpmyadminPort`) and environment variables (`WP_ENV_PHPMYADMIN_PORT`, `WP_ENV_TESTS_PHPMYADMIN_PORT`).


## Testing Instructions
* Run `WP_ENV_PHPMYADMIN_PORT=9000 npx wp-env start`.
* Open `localhost:9000`.
* Verify that phpMyAdmin is available and correctly connected to the `mysql` database service.

* Repeat, but, instead of passing `WP_ENV_PHPMYADMIN_PORT=9000`, edit your `.wp-env.json` file to add the following key-value pair under `env > development` and just run `npx wp-env start`:

```json
{
	"env": {
		"development": {
			"phpmyadminPort": 9000
```

* Repeat but with the testing environment:
  * Try with `WP_ENV_TESTS_PHPMYADMIN_PORT=9003 npx wp-env start` and check that the `tests-phpmyadmin` is correctly launched.
  * Try adding the config under `env > tests` and check that the `tests-phpmyadmin` is correctly launched.

## Checklist

- [X] Question: Is it worth simplifying by, instead of loading port configuration via the config system, just require the port to be provided in the `start` invocation: `npx wp-env start --phpmyadmin=9000`?
- [X] What's the best way to extend this to the test database?
